### PR TITLE
Support external spire

### PR DESCRIPTION
**Pull request by danielpcox**
[sc-13415]

Companion PR in gitops-core: https://github.com/greymatter-io/gitops-core/pull/73

### CODE REVIEW

File: pkg/cuemodule/load.go [hunk 1]
```diff
@@ -75,6 +76,11 @@ type Config struct {
 	OperatorNamespace  string `json:"operator_namespace"`
 }
 
+type Spire struct {
```
Comment (dontlaugh) [inline pkg/cuemodule/load.go:79]: This holds the config of a K8s secret's 1) name and 2) namespace, right? Should we call this (something like) `SpireSecret`?

In any case, add a doc string to all exported members in Go files.

Comment (danielpcox) [reply pkg/cuemodule/load.go:79]: No, it's not actually a secret, and that's not all it holds. It's a miscellaneous bag of stuff necessary to point at an existing spire installation, and the actual block is named `defaults.spire` so I'd prefer to call it "spire" to match. It doesn't need to be exported though, so I lowercased it, and I added a doc string to the parent `Defaults` struct.

Comment (dontlaugh) [reply pkg/cuemodule/load.go]: unexporting it is even better

**Review by dontlaugh — DISMISSED** [review]

**Review by dontlaugh — APPROVED** [review]

**Comment by dontlaugh** [comment]
Good to merge if the tests are green.

**Pull request merged**